sdk/servicebus/azure-servicebus/azure/servicebus/_common/_connection_string_parser.py

```diff
@@ -0,0 +1,88 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse  # type: ignore
+
+from azure.servicebus.management._models import DictMixin
+
+class ServiceBusConnectionStringProperties(DictMixin):
+    """
+    Properties of a connection string.
+    """
+    def __init__(self, **kwargs):
+        self._fully_qualified_namespace = kwargs.pop('fully_qualified_namespace', None)
+        self._endpoint = kwargs.pop('endpoint', None)
+        self._entity_path = kwargs.pop('entity_path', None)
+        self._shared_access_signature = kwargs.pop('shared_access_signature', None)
+        self._shared_access_key_name = kwargs.pop('shared_access_key_name', None)
+        self._shared_access_key = kwargs.pop('shared_access_key', None)
+
+    @property
+    def fully_qualified_namespace(self):
+        """The fully qualified host name for the Service Bus namespace.
+        The namespace format is: `<yournamespace>.servicebus.windows.net`.
+        """
+        return self._fully_qualified_namespace
+
+    @property
+    def endpoint(self):
+        """The endpoint for the Service Bus resource. In the format sb://<FQDN>/
+        """
+        return self._endpoint
+
+    @property
+    def entity_path(self):
+        """Optional. Represents the name of the queue/topic.
+        """
+        return self._entity_path
+
+    @property
+    def shared_access_signature(self):
```

Comment:
oops, I found `shared_access_key_name` and `shared_access_key` are missing, should we add them too?